contracts/container/contract.go

```diff
@@ -1297,6 +1307,139 @@ func NewEpoch(epochNum int) {
 	cleanupContainers(ctx, epochNum)
 }
 
+// PutContainerQuota sets size quota that limits all space used for storing
+// objects in cID (including object replicas). Non-positive size sets no
+// limitation. if hard is true, storage nodes decline any exceeding object at
+// PUT operation; if false, nodes are instructed to warn only, without denial
+// of service. Call must be signed by cID's owner. Limit can be changed with
+// repeated call.
+// Panics if cID is incorrect, container does not exist or new limitation already
+// exceeded (according to registered NodeReports).
+func PutContainerQuota(cID interop.Hash256, size int, hard bool) {
+	if len(cID) != interop.Hash256Len {
+		panic("wrong container id")
+	}
+	ctx := storage.GetContext()
+	ownerID := getOwnerByID(ctx, cID)
+	if ownerID == nil {
+		panic(cst.NotFoundError)
+	}
+
+	ownerSH := common.WalletToScriptHash(ownerID)
+	common.CheckOwnerWitness(ownerSH)
+
+	netmapContractAddr := storage.Get(ctx, netmapContractKey).(interop.Hash160)
+	currEpoch := contract.Call(netmapContractAddr, "epoch", contract.ReadOnly).(int)
+
+	if size > 0 {
+		var currentSize int
+		it := IterateReports(currEpoch, cID)
+		for iterator.Next(it) {
+			r := iterator.Value(it).(NodeReport)
+			currentSize += r.ContainerSize
+		}
+		if currentSize >= size {
+			panic(cst.ErrorQuotaLimitAlreadyExceeded)
+		}
+	}
+
+	key := []byte{containerQuotaKeyPrefix}
+	key = append(key, cID...)
```

Comment:
Usually we move such code to helpers like `getContainerQuotaKey` and reuse from getter/setter, but it's up to you.